src/getuuid.c

```diff
@@ -41,15 +85,24 @@ int getuuidn(char *buff, const char *id, size_t len)
 
   if (!id || !*id) {
     int status = uuid4_generate(buff);
-    version = (status == 0) ? 4 : -1;
-  } else if (uuid_from_string(NULL, id, len)) {
-    uuid_create_sha1_from_name(&uuid, NameSpace_DNS, id, len);
-    uuid_as_string(&uuid, buff);
-    version = 5;
-  } else {
+    if (status) return -1;
+    version = UUID_RANDOM;
+  } else if (((len == UUID_LEN) ||
+              (len == UUID_LEN+1 && strchr("#/", id[len]))) &&
+             isuuid(id)) {
     strncpy(buff, id, UUID_LEN);
     buff[UUID_LEN] = '\0';
-    version = 0;
+    version = UUID_COPY;
+  } else if (isinstanceuri(id, len)) {
+    if (id[len-1] && strchr("#/", id[len-1])) len--;
+    len -= UUID_LEN;
+    assert(len > 0);
```

Comment:
Why assert here?

Reply:
It is just a safety guard. If the code works as intended, `len` should always be tositive. 

If `id` is an instance URI of the form `<meta.uri>/<uuid>` then `len` should be positive (larger than 9, I think). What I really don't want is that `len` for some unknown reason becomes negative, resulting in accessing some random memory. Then I prefer aborting dlite loudly...